lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -93,63 +93,6 @@ class SetVariantSelectionUndoableCommand : public Ufe::UndoableCommand
     const std::string fNewSelection;
 };
 
-#if UFE_PREVIEW_VERSION_NUM < 2017
-//! \brief Undoable command for add new prim
-class AddNewPrimUndoableCommand : public Ufe::UndoableCommand
-{
-public:
-    AddNewPrimUndoableCommand(const MayaUsd::ufe::UsdSceneItem::Ptr& usdSceneItem,
-                              const Ufe::ContextOps::ItemPath& itemPath)
-    : Ufe::UndoableCommand()
-    {
-        // First get the stage from the proxy shape.
-        auto ufePath = usdSceneItem->path();
-        auto segments = ufePath.getSegments();
-        auto dagSegment = segments[0];
-        _stage = MayaUsd::ufe::getStage(Ufe::Path(dagSegment));
-        if (_stage) {
-            // Rename the new prim for uniqueness, if needed.
-            // When we have a gateway type node, the prim we'll be adding will be
-            // in a different runtime (USD vs Maya). So technically just appending
-            // itemPath[1] isn't correct. But in this case it is only used by
-            // uniqueChildName() which just takes the back of the path for child
-            //  name, so it's fine.
-            Ufe::Path newUfePath = ufePath + (itemPath[1] + std::to_string(1));
-            auto newPrimName = uniqueChildName(usdSceneItem, newUfePath);
-
-            // Build (and store) the path for the new prim with the unique name.
-            PXR_NS::SdfPath usdItemPath = usdSceneItem->prim().GetPath();
-            _primPath = usdItemPath.AppendChild(PXR_NS::TfToken(newPrimName));
-
-            // The type of prim we were asked to create.
-            // Note: "Def" means create typeless prim.
-            _primToken = (itemPath[1] == kUSDDefPrimItem) ? TfToken() : TfToken(itemPath[1]);
-        }
-    }
-
-    void undo() override
-    {
-        if (_stage) {
-            _stage->RemovePrim(_primPath);
-        }
-    }
-
-    void redo() override
-    {
-        if (_stage) {
-            auto newPrim = _stage->DefinePrim(_primPath, _primToken);
-            if (!newPrim.IsValid())
-                TF_RUNTIME_ERROR("Failed to create new prim type: %s", _primToken.GetText());
-        }
-    }
-
-private:
-    PXR_NS::UsdStageWeakPtr _stage;
-    PXR_NS::SdfPath _primPath;
-    PXR_NS::TfToken _primToken;
-};
-#endif
```

Comment:
Removed duplicated AddNewPrimUndoableCommand since we now are using UsdUndoAddNewPrimCommand.

Reply:
Not really a duplicate, it's for UFE versions before 0.2.17.

Reply:
@ppt-adsk Correct me if I am wrong but I don't see anything in the removed code that relied on older version of UFE. 

Almost all of code in AddNewPrimUndoableCommand got duplicated in the new UsdUndoAddNewPrimCommand class with a few logic changes which are not really relied on UFE interface changes.

Keeping AddNewPrimUndoableCommand  and UsdUndoAddNewPrimCommand  together is very confusing and hard to maintain.

Reply:
Yes, Hamed and I talked about this one yesterday.  I think when I was first going through the code updating to 2.17 I was wrapping everything in the ifdef's, but this one could have been removed entirely.  There was my fix to not use the name as the type when creating a group, but that doesn't require us to keep the old version around.

Reply:
O.K., you guys know what you're talking about.